rfcs/min-relations.md

```diff
@@ -238,24 +248,123 @@ fn reattach_springs(mut commands: Commands,
 
 ### Advanced relation filters
 
-TODO: Complete examples in this section
+The examples shown above show of the basics of relation filtering, but that's not all they can do.
+The most important extension is the ability to filter for multiple entities at once from a `Vec<Entity>`:
 
-For those cases where previous examples weren't enough, there are several additional API extensions that you might want to use. Using those, Relation filters can be combined in arbitrarily complex ways.
+```rust
+// In this example, possible movess are stored on each piece
+// as a relation to a specific board position
+fn next_move(mut commands: Commands,
+             board_state: Res<BoardState>,
+             query: Query<(Entity, &Relation<Move>), With<Piece>>){    
+    
+    let valid_positions: Vec<Entity> = board_state.compute_valid_positions();
 
-The first of these is the ability to filter by the source entity, rather than target entity.
-Note that you can get the same sort of effect by using `Query::get`;
-this functionality just makes it more ergonomic to specify complex relation filters.
+    let filtered = query.filter_relation::<Action, _>( 
+        RelationFilter::any_of().targets(valid_positions)
+    ).build();
 
-```rust
-fn 
+    for (piece, valid_moves) in filtered.iter(){
+        for (potential_position, current_move) in valid_moves {
+            // Our heuristic is always >= 0
+            let mut current_best = -1.0;
+            // This is just a dummy to be replaced
+            let best_move = Entity::new();
+            let current = current_move.compute_heuristic()
+
+            if current > current_best{
+                current_best = current;
+                best_move = current_move;
+            }
+        }
+        commands.insert_relation::<NextMove>(piece, best_move);
```

Comment:
this `for` loop could be replaced with simple `.max_by_key(..)`;
```suggestion
		let best_move = valid_moves
			.max_by_key(|(potential_position, _)| potential_position.compute_heuristic());
			
		if let Some((_, best_move)) = best_move {
			commands.insert_relation::<NextMove>(piece, best_move);
		}
```

also changed `compute_heuristic` to be a method of `potential_position`, as it made more sense to me and also uses both tuple arguments. Please correct me if that's wrong for some reason.

Reply:
Awesome. I agree with all of those changes.

Reply:
Applied now.

Reply:
This change has been applied <3